lib/bmp280.ex

```diff
@@ -136,27 +139,16 @@ defmodule BMP280 do
       state
       |> query_sensor()
       |> init_sensor()
+      |> read_and_put_new_measurement()
+
+    schedule_measurement()
 
     {:noreply, new_state}
   end
 
   @impl GenServer
   def handle_call(:measure, _from, state) do
-    rc =
-      case read_raw_samples(state.transport, state.sensor_type) do
-        {:ok, raw} ->
-          {:ok,
-           Calc.raw_to_measurement(
-             state.calibration,
-             state.sea_level_pa,
-             raw
-           )}
-
-        error ->
-          error
-      end
-
-    {:reply, rc, state}
+    {:reply, state.last_measurement, state}
```

Comment:
```suggestion
    if state.last_measurement do
      {:reply, {:ok, state.last_measurement}, state}
    else
      {:reply, {:error, :no_measurement}, state}
    end
```
It looks like you want --^